front/pages/api/w/[wId]/subscriptions/stripe_subscription_valid.ts

```diff
@@ -0,0 +1,188 @@
+import type { WithAPIErrorReponse } from "@dust-tt/types";
+import type { NextApiRequest, NextApiResponse } from "next";
+import type Stripe from "stripe";
+
+import { Authenticator, getSession } from "@app/lib/auth";
+import { getStripeSubscription } from "@app/lib/plans/stripe";
+import { apiError, withLogging } from "@app/logger/withlogging";
+
+// TODO(2024-04-05,pr): immediately after flav's merge, use the global constant
+const REPORT_USAGE_VALUES = ["MAU_1", "MAU_5", "MAU_10", "PER_SEAT"];
+
+export type GetSubscriptionResponseBody =
+  | { valid: true }
+  | { valid: false; reason: string };
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorReponse<GetSubscriptionResponseBody>>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const stripeSubscriptionId = req.query.stripeSubscriptionId as string;
+  if (!stripeSubscriptionId || typeof stripeSubscriptionId !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid request, missing stripeSubscriptionId.",
+      },
+    });
+  }
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  // Callers will be either superUsers (from poke) or admins of the workspace (from webhook calls)
+  if (!auth.isAdmin()) {
```

Comment:
IMO, this should be the first few lines of the endpoint. Before we even check query parameters.